src/newViews/Tagging/Sources/LeadEditModal/LeadEditForm/index.tsx

```diff
@@ -139,6 +195,122 @@ function LeadEditForm(props: Props) {
+                if (webInfo.source) {
+                    // eslint-disable-next-line no-param-reassign
+                    safeValues.source = webInfo.source.id;
+                }
+                if (webInfo.author) {
+                    // FIXME: we have to look into this
+                    // eslint-disable-next-line no-param-reassign
+                    safeValues.authors = [webInfo.author.id];
+                }
+            });
+            return newValues;
+        });
+        if (webInfo.source) {
+            setSourceOrganizationOptions(
+                oldVal => [...oldVal ?? [], webInfo.source].filter(isDefined),
+            );
+        }
+        if (webInfo.author) {
+            setAuthorOrganizationOptions(
+                oldVal => [...oldVal ?? [], webInfo.author].filter(isDefined),
+            );
+        }
+        setPristine(false);
+    }, [setValue, setPristine]);
+
+    const {
+        pending: webInfoPending,
+        trigger: getWebInfo,
+    } = useLazyRequest<WebInfo, WebInfoBody>({
+        method: 'POST',
+        url: 'server://v2/web-info-data/',
+        body: ctx => ctx,
+        onSuccess: (response, ctx) => {
+            handleInfoAutoFill({
+                date: ctx.date,
+                website: ctx.website,
+                title: ctx.title,
+                url: ctx.url,
+                ...response,
+            });
+        },
+        failureHeader: 'Web Info Extract',
+    });
+
+    const {
+        pending: rawWebInfoPending,
+        trigger: getRawWebInfo,
+    } = useLazyRequest<RawWebInfo, { url: string; isFile: boolean }>({
+        method: 'GET',
+        url: 'serverless://web-info-extract/',
+        query: ctx => ({ url: ctx.url }),
+        onSuccess: (response, ctx) => {
+            if (ctx.isFile) {
+                handleInfoAutoFill({
+                    title: response.title,
+                });
+            } else {
+                getWebInfo({
+                    url: ctx.url,
+                    title: response.title,
+                    date: response.date,
+                    website: response.website,
+                    country: response.country,
+                    sourceRaw: response.sourceRaw,
+                    authorRaw: response.authorRaw,
+                });
+            }
+        },
+        failureHeader: 'Raw Web Info Extract',
+    });
+
+    const {
+        pending: pendingUserToken,
+        trigger: getUserToken,
+    } = useLazyRequest<Token, { isFile: boolean }>({
+        method: 'GET',
+        url: 'server://token/',
+        onSuccess: (_, ctx) => {
+            if (ctx.isFile && value.attachment?.file) {
+                getRawWebInfo({
+                    url: value.attachment.file,
+                    isFile: true,
+                });
+            } else if (value.url) {
+                getRawWebInfo({
+                    url: value.url,
+                    isFile: false,
+                });
+            }
```

Comment:
What if it's something else? Let's add a console.error if there shouldn't be such a case.

Reply:
Done